tests/compiler/function-call.release.wat

```diff
@@ -57,14 +57,15 @@
  (export "memory" (memory $0))
  (start $~start)
  (func $start:function-call~anonymous|0
+  nop
  )
  (func $start:function-call~anonymous|2 (param $0 i32) (param $1 i32) (result i32)
   local.get $0
   local.get $1
   i32.add
  )
  (func $start:function-call~fn2|4 (param $0 i32) (result i32)
-  i32.const 1
+  local.get $0
```

Comment:
Hmm, this definitely got worse.

Reply:
I believe these two cases are caused by the same issue (the `dae`/DeadArgumentElimination pass not doing its job? I'm not fully sure).

I did a bisection to find the Binaryen commit that's responsible. (I tested on the `call-optional` WAT file in S-expr format, but the issue applies here as well.) It looks like WebAssembly/binaryen#7135 is the cause, as that PR treats any funcref as being possibly leaked to the outside world...

How should we proceed? I don't think marking the module as closed-world is accurate...

Reply:
@MaxGraey Should we just eat the cost and merge this PR? We could also mark the module as closed world if functions don't return funcrefs and tables aren't imported/exported...what do you think?

Reply:
> We could also mark the module as closed world if exported functions don't return funcrefs and tables aren't imported/exported...what do you think?

That would be ideal because it's a quite common scenario.

Reply:
@MaxGraey I made that change now!

(Now that I think about it, AS currently never exports arrow functions/etc. as funcrefs, but as boxed objects instead...regardless, the change accounts for table imports/exports as well as users specifying `ref_func` exports, or exported functions returning `ref_func`.)